### PR TITLE
matchers: Fix matching on multi-lines comments

### PR DESCRIPTION
**Pull request by apelisse**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1655)

### CODE REVIEW

**Comment by foxish** [comment]
@apelisse  Could you explain what was wrong with the previous regexes and what the issue was? 
I'm trying to understand what happened, and it'll help review this :)

File: mungegithub/mungers/matchers/comment/notification.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ type Notification struct {
 
 var (
 	// Matches a notification: [NOTIFNAME] Arguments
-	notificationRegex = regexp.MustCompile(`^\[([^\]\s]+)\]\s?(.*)$`)
+	notificationRegex = regexp.MustCompile(`(?m:^\[([^\]\s]+)\] *?(.*)$)`)
```
Comment (foxish) [inline mungegithub/mungers/matchers/comment/notification.go]:
```
(?m:^\[([^\]\s]+)\] *?(.*)$)
-------^ do we need this capturing group?
```

Comment (foxish) [inline mungegithub/mungers/matchers/comment/notification.go]: Also why do we explicitly include `\s`? Wouldn't `[^\]]` include spaces anyway?

File: mungegithub/mungers/matchers/comment/command_test.go [hunk 2]
```diff
@@ -45,6 +45,18 @@ func TestParseCommand(t *testing.T) {
 			comment:         "/COMMAND Valid command",
 		},
 		{
+			expectedCommand: &Command{Name: "COMMAND", Arguments: "Multiple Lines"},
```
Comment (foxish) [inline mungegithub/mungers/matchers/comment/command_test.go]: Do we want to support this syntax of the command not being at the beginning of a comment?

File: mungegithub/mungers/matchers/comment/notification.go (hunk 1 above)
Comment (apelisse) [inline mungegithub/mungers/matchers/comment/notification.go]: - Do we need to capture this group: Yes, it's the name of the notification
- Why do we include \s twice: No, we are actually excluding \s (confusing with all those squared brackets ...)

File: mungegithub/mungers/matchers/comment/command_test.go (hunk 2 above)
Comment (foxish) [inline mungegithub/mungers/matchers/comment/command_test.go]: @grodrigues3 WDYT?

Comment (grodrigues3) [inline mungegithub/mungers/matchers/comment/command_test.go]: I'd like commands to be at the beginning of comments.  Easier for humans to parse and figure out where bot commands came from.

**Comment by foxish** [comment]
LGTM

**Comment by grodrigues3** [comment]
LGTM

**Pull request merged**